app/models/comment.rb

```diff
@@ -0,0 +1,10 @@
+class Comment < ActiveRecord::Base
+  has_many :comments, as: :commentable
+  belongs_to :commentable
+  belongs_to :author, class_name: "User"
+
+
+  validates :body, presence: true, length: { minimum: 2, maximum: 160 }
+  validates_presence_of :author_id
```

Comment:
validates_presence_of :author would be better - it would check that author_id is a valid id for a user, not just that it's not null.